ios/Sources/AutoLayoutViewComponentView.mm

```diff
@@ -0,0 +1,80 @@
+#ifdef RN_FABRIC_ENABLED
+#import "AutoLayoutViewComponentView.h"
+#import <React/RCTConversions.h>
+#import <React/RCTViewManager.h>
+
+#import <react/renderer/components/rnflashlist/ComponentDescriptors.h>
+#import <react/renderer/components/rnflashlist/EventEmitters.h>
+#import <react/renderer/components/rnflashlist/Props.h>
+#import <react/renderer/components/rnflashlist/RCTComponentViewHelpers.h>
+
+#import "RCTFabricComponentsPlugins.h"
+#import <RNFlashList-Swift.h>
+
+using namespace facebook::react;
+
+@interface AutoLayoutViewComponentView () <RCTAutoLayoutViewViewProtocol>
+@end
+
+@implementation AutoLayoutViewComponentView
+{
+    AutoLayoutView *_autoLayoutView;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+  if (self = [super initWithFrame:frame]) {
+    static const auto defaultProps = std::make_shared<const AutoLayoutViewProps>();
+    _props = defaultProps;
+    _autoLayoutView = [[AutoLayoutView alloc] initWithFrame:self.bounds];
+
+    // Due to view flattening, AutoLayoutView's children get moved to its parent (AutoLayoutViewComponentView) and
+    // AutoLayoutView is positioned above them consuming all events. Turning off userInteraction prevents that.
+    _autoLayoutView.userInteractionEnabled = false;
+
+    self.contentView = _autoLayoutView;
+
+    __weak AutoLayoutViewComponentView* weakSelf = self;
+    _autoLayoutView.onBlankAreaEventHandler = ^(CGFloat start, CGFloat end) {
+      AutoLayoutViewComponentView *strongSelf = weakSelf;
+      if (strongSelf->_eventEmitter != nullptr) {
+        std::dynamic_pointer_cast<const facebook::react::AutoLayoutViewEventEmitter>(strongSelf->_eventEmitter)
+          ->onBlankAreaEvent(facebook::react::AutoLayoutViewEventEmitter::OnBlankAreaEvent{
+            .offsetStart = (int) floor(start),
+            .offsetEnd = (int) floor(end),
+          });
+      }
```

Comment:
Can you link me some additional resources? This seems quite complicated for sending an event, so would love to learn more.